file/random_access_file_reader.cc

```diff
@@ -276,6 +276,13 @@ IOStatus RandomAccessFileReader::Read(const IOOptions& opts, uint64_t offset,
     IOStatsAddCountByTemperature(file_temperature_, 1);
     StatisticAddBytesByTemperature(stats_, file_temperature_, result->size());
     StatisticAddCountByTemperature(stats_, file_temperature_, 1);
+    if (is_last_level_) {
```

Comment:
I don't like this if. This will create branch misprediction in a critical path. Although this is unlikely to be the bottleneck but small things might add up. Is it possible to remember the two tick names when creating RandomAccessFileReader?

Reply:
It's actually the same as the temperature checks above. I'll see if we can optimize them.

Reply:
I updated that part a little bit, basically grouping the temperature checking, which reduced the temperature if checking 3 times even when it's not enabled. I added a microbenchmark test, which shows the performance is even improved when statistics not enabled:
```
Main:     255ns
With fix: 252ns
Old fix:  256ns
```
When statistics is enabled, the time is increased because more statistics recording, which is unavoidable for more statistics:
```
Main:     340ns
With fix: 370ns
Old fix:  374ns
```